CONTRIBUTING.md

```diff
@@ -143,10 +143,37 @@ and the rest of contributors.
   and in the respective testing harnesses. To align with the test goal of the project, components must be testable within the framework defined within
   the folder. If a component can not be properly tested within the existing framework, it must increase the non testable
   components number with a comment within the PR explaining as to why it can not be tested.
-- Add the sponsor for your component and yourself to a new line for your component in the
-  [`.github/CODEOWNERS`](./.github/CODEOWNERS) file.
+- Create a `metadata.yaml` file with at minimum the required fields defined in [metadata-schema.yaml](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/cmd/mdatagen/metadata-schema.yaml).
+Here is a minimal representation:
+```
+type: <name of your component, such as apache, http, haproxy, postgresql>
+
+status:
+  class: <class of component, one of cmd, connector, exporter, extension, processor or receiver>
+  stability:
+    development: [<pick the signals supported: logs, metrics, traces. For extension, use "extension">]
+  codeowners:
+    active: [<github account of the sponsor, such as alice>, <your GitHub account if you are already an OpenTelemetry member>]
+```
+- Run `make gengithub` to generate the updated `.github/CODEOWNERS` file.
 - Run `make generate-gh-issue-templates` to add your component to the dropdown list in the issue templates.
-- Create a `metadata.yaml` file with at minimum the required fields defined in [metadata-schema.yaml](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/cmd/mdatagen/metadata-schema.yaml) and use the [metadata generator](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/cmd/mdatagen/README.md#using-the-metadata-generator) to generate the associated code/documentation.
+- For README.md, you can start with the following:
+```
+# <Title of your component>
+<!-- status autogenerated section -->
+<!-- end autogenerated section -->
+```
+- Create a `doc.go` file with a generate pragma. For a `fooreceiver`, the file will look like:
+```
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+//go:generate mdatagen metadata.yaml
+
+// Package fooreceiver bars.
+package fooreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/fooreceiver"
+```
+- Type `make generate`. This will trigger the [metadata generator](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/cmd/mdatagen/README.md#using-the-metadata-generator) to generate the associated code/documentation.
```

Comment:
```suggestion
- Type `make update-codeowners`. This will trigger the regeneration of the `.github/CODEOWNERS` file and the [metadata generator](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/cmd/mdatagen/README.md#using-the-metadata-generator) to generate the associated code/documentation.
```